### PR TITLE
[chore]move googlecloudpubsubreceiver to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27850

### CODE REVIEW

File: receiver/googlecloudpubsubreceiver/metadata.yaml [hunk 1]
```diff
@@ -7,3 +7,11 @@ status:
...
   codeowners:
     active: [alexvanboxel]
+
+tests:
+  config:
+    project: my-project
+    user_agent: user-agent
+    timeout: 20s
+    subscription: projects/my-project/subscriptions/otlp-subscription
+  skip_lifecycle: true
```
Comment (codeboten) [inline receiver/googlecloudpubsubreceiver/metadata.yaml:17]: add issue to not skip lifecycle tests

File: receiver/googlecloudpubsubreceiver/receiver.go [hunk 2]
```diff
@@ -111,6 +111,9 @@ func (receiver *pubsubReceiver) Start(ctx context.Context, _ component.Host) err
 }
 
 func (receiver *pubsubReceiver) Shutdown(_ context.Context) error {
+	if receiver.handler == nil {
+		return nil
+	}
 	receiver.logger.Info("Stopping Google Pubsub receiver")
 	receiver.handler.CancelNow()
```
Comment (codeboten) [inline receiver/googlecloudpubsubreceiver/receiver.go:118]: the check for nil might be better around this call to still have the logs?

Comment (atoulme) [reply receiver/googlecloudpubsubreceiver/receiver.go:118]: I don’t think we are stopping anything if we didn’t start, so no logging might be best. Codeowners might know best.

Comment (alexvanboxel) [reply receiver/googlecloudpubsubreceiver/receiver.go:118]: Indeed, the handler is the thing that needs to be stopped and will consume resources. So nothing needs to be logged.

**Review by alexvanboxel — APPROVED** [review]

**Comment by alexvanboxel** [comment]
lgtm

**Review by codeboten — APPROVED** [review]

**Pull request merged**